src/igvfd/schemas/treatment.json

```diff
@@ -128,34 +127,95 @@
-                                "protein"
-                            ]
+                    "if": {
+                        "properties": {
+                            "treatment_type": {
+                                "const": "protein"
+                            }
                         },
-                        "treatment_term_id": {
-                            "pattern": "^((UniProtKB:[A-Z0-9]{6})|(NTR:[0-9]{2,8}))$"
-                        }
+                        "required": [
+                            "treatment_type"
+                        ]
+                    },
+                    "then": {
+                        "properties": {
+                            "treatment_term_id": {
+                                "pattern": "^((UniProtKB:[A-Z0-9]{6})|(NTR:[0-9]{2,8}))$"
+                            }
+                        },
+                        "required": [
+                            "treatment_term_id"
+                        ]
                     }
                 },
                 {
-                    "properties": {
-                        "treatment_type": {
-                            "enum": [
-                                "chemical"
-                            ]
+                    "if": {
+                        "properties": {
+                            "treatment_type": {
+                                "const": "chemical"
+                            }
+                        },
+                        "required": [
+                            "treatment_type"
+                        ]
+                    },
+                    "then": {
+                        "properties": {
+                            "treatment_term_id": {
+                                "pattern": "^((CHEBI:[0-9]{1,7})|(NTR:[0-9]{2,8}))$"
+                            }
                         },
-                        "treatment_term_id": {
-                            "pattern": "^((CHEBI:[0-9]{1,7})|(NTR:[0-9]{2,8}))$"
+                        "required": [
+                            "treatment_term_id"
+                        ]
+                    }
+                },
+                {
+                    "if": {
+                        "properties": {
+                            "treatment_type": {
+                                "const": "environmental"
+                            }
+                        },
+                        "required": [
+                            "treatment_type"
+                        ]
+                    },
+                    "then": {
+                        "not": {
+                            "required": [
+                                "treatment_term_id"
+                            ]
                         }
                     }
                 }
             ]
+        },
+        "treatment_term_name": {
+            "comment": "Environmental treatments are restricted to certain term names including: stiffness.",
```

Comment:
Maybe "restricted to term names that include currently only stiffness."

Reply:
done.